src/hdmf/container.py

```diff
@@ -85,10 +86,78 @@ def setter(self, val):
             if name in self.fields:
                 msg = "can't set attribute '%s' -- already set" % name
                 raise AttributeError(msg)
-            self.fields[name] = val
+            self.fields[name] = self._field_config(arg_name=name, val=val)
 
         return setter
 
+
+    def _field_config(self, arg_name, val):
+        """
+        This method will be called in the setter. The termset configuration will be used (if loaded)
+        to check for a defined TermSet associated with the field. If found, the value of the field
+        will be wrapped with a TermSetWrapper.
+
+        Even though the path field in the configurator can be a list of paths, the config
+        itself is only one file. When a user loads custom configs, the config is appended/modified.
+        The modifications are not written to file, avoiding permanent modifications.
+        """
+        # load termset configuration file from global Config
+        try:
+            configurator = self.type_map.ts_config
+            if len(configurator.path)>0:
+                # The type_map has a config always set; however, when toggled off, the config path is empty.
+                CUR_DIR = os.path.dirname(os.path.realpath(configurator.path[0]))
+                termset_config = configurator.config
+            else:
+                return val
+        except AttributeError:
+            # These are for classes that have not been registered, e.g., some test containers.
+            # Without register_class, the type_map parameter is not defined.
+            return val
+
+        # check to see that the namespace for the container is in the config
+        if self.namespace not in self.type_map.container_types:
+            msg = "%s not found within loaded configuration." % self.namespace
+            warn(msg)
+            return val
+        else:
+            # check to see that the container type is in the config under the namespace
+            config_namespace = termset_config['namespaces'][self.namespace]
+            object_name = self.__class__.__name__
```

Comment:
Because of how `register_class` works, the class name is not guaranteed to be the same as the data type name. You can get the type name through `getattr(self, self._data_type_attr)`. It would probably be useful to put that into its own `get_data_type` method.

Reply:
Also I suggest changing the variable name from `object_name` to `class_name`. I would interpret the name of the object as the name of the container.

Reply:
Can do, but what case would they be different and why do we do that?